src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -15,8 +14,8 @@ internal static class DocumentExtensions
     {
         public static bool ShouldHideAdvancedMembers(this Document document)
         {
-            var service = document.Project.Solution.Workspace.Services.GetService<IOptionService>();
-            return service.GetOption(CompletionOptions.HideAdvancedMembers, document.Project.Language);
+            return document.Project.Solution.Workspace.Options
```

Comment:
You're right. I'll update GetUpdatedRecommendationOptions. This code is a bit deadly.